app/src/main/java/com/github/warnastrophy/core/data/repository/NominatimRepository.kt

```diff
@@ -120,33 +118,27 @@ class NominatimRepository(private val ioDispatcher: CoroutineDispatcher = Dispat
    * @param urlStr The full URL to fetch.
    * @return The response body as a string, or an empty string if rate limited.
    */
-  private suspend fun httpGet(urlStr: String): String =
-      withContext(ioDispatcher) {
-        val bool = isRateLimited()
-
-        if (bool) {
-          return@withContext ""
-        }
-
-        val url = URL(urlStr)
-        val conn =
-            (url.openConnection() as HttpURLConnection).apply {
-              requestMethod = "GET"
-              setRequestProperty("Accept-Language", "en")
-              setRequestProperty("User-Agent", "WarnAStrophyApp/1.0 (+${referer}")
-              setRequestProperty("Referer", referer)
-            }
-        val message: String
-        try {
-          val code = conn.responseCode
-          val stream = if (code in 200..299) conn.inputStream else conn.errorStream
-          message = BufferedReader(InputStreamReader(stream)).use { it.readText() }
-        } finally {
-          conn.disconnect()
+  private fun httpGet(urlStr: String): String {
+    lastQueryTimestamp = System.currentTimeMillis()
+    val url = URL(urlStr)
+    val conn =
+        (url.openConnection() as HttpURLConnection).apply {
+          requestMethod = "GET"
+          setRequestProperty("Accept-Language", "en")
+          setRequestProperty("User-Agent", "WarnAStrophyApp/1.0 (+$referer)")
+          setRequestProperty("Referer", referer)
         }
-        return@withContext message
-      }
 
+    return try {
+      val code = conn.responseCode
+      val stream = if (code in 200..299) conn.inputStream else conn.errorStream
```

Comment:
As it is done in `HazardsRepository`, you could introduce  a range to give meaning to these _magic numbers_: 

`private val VALID_REPONSE_CODE_RANGE = 200..299`

Then:

```suggestion
      val stream = if (code in VALID_REPONSE_CODE_RANGE) conn.inputStream else conn.errorStream
```

Reply:
done